### PR TITLE
Action/Handler fixes

### PR DESCRIPTION
**Pull request by kalantar**
Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>

Fixes https://github.com/iter8-tools/iter8/issues/982 by fixing IsExperimentValid() and https://github.com/iter8-tools/iter8/issues/983 by changing the way task jobs are started.

### CODE REVIEW

File: controllers/config.go [hunk 1]
```diff
@@ -66,9 +61,9 @@ func (b Iter8ConfigBuilder) WithNamespace(namespace string) Iter8ConfigBuilder {
 	return b
 }
 
-// WithHandlersDir ..
-func (b Iter8ConfigBuilder) WithHandlersDir(handlersDir string) Iter8ConfigBuilder {
-	b.HandlersDir = handlersDir
+// WithTasRunner ..
```
Comment (sriumcp) [inline controllers/config.go:64]: // WithTaskRunner (k)

File: controllers/config_test.go [hunk 2]
```diff
@@ -36,18 +36,18 @@ func TestNoInterpolation(t *testing.T) {
...
+		t.Errorf("cfg.TaskRunner incorrect. Expected: %s, got: %s", "taskrunner", cfg.TaskRunner)
 	}
-	if cfg.Analytics.Endpoint != "endpoint" {
-		t.Errorf("cfg.Analytics.Endpoint incorrect. Expected: %s, got: %s", "endpoint", cfg.Analytics.Endpoint)
+	if cfg.AnalyticsEndpoint != "endpoint" {
+		t.Errorf("cfg.Endpoint incorrect. Expected: %s, got: %s", "endpoint", cfg.AnalyticsEndpoint)
 	}
 }
 
 func TestInterpolation(t *testing.T) {
```
Comment (sriumcp) [inline controllers/config_test.go:47]: Why are these tests named TestInterpolation and TestNoInterpolation?

File: controllers/handler.go [hunk 3]
```diff
@@ -110,41 +108,15 @@ func (r *ExperimentReconciler) LaunchHandler(ctx context.Context, instance *v2be
...
-	// 	JobServiceAccountName: "default",
-	// 	Image:                 "iter8/iter8-kfserving:latest",
-	// 	Commands:              []string{"handlers/scripts/start.sh"},
-	// 	ExperimentName:        instance.Name,
-	// 	ExperimentNamespace:   instance.Namespace,
-	// })
+	job := defineJob(jobHandlerConfig{
+		JobName:             jobName(instance, handler, handlerInstance),
+		JobNamespace:        r.Iter8Config.Namespace,
+		Image:               r.Iter8Config.TaskRunner,
```
Comment (sriumcp) [inline controllers/handler.go:114]: Where is the image value coming from for task runner? I don't see it on config or config_test.

Comment (sriumcp) [reply controllers/handler.go:114]: Actually, I assume the `TaskRunner` env variable is intended to be the `TaskRunner` image?

If so, could you please append `Image` to it to make the naming explicit?

**Review by sriumcp — CHANGES_REQUESTED** [review]
Please see review comments.

File: controllers/config.go (hunk 1 above)
Comment (kalantar) [reply controllers/config.go]: Fixed name in comment; added better comments.

File: controllers/config_test.go (hunk 2 above)
Comment (kalantar) [reply controllers/config_test.go]: The analytics endpoint can be expressed using the name ITER8_NAMESPACE which is interpolated using the value of this env variable. These tests test each case. Renamed to be clearer. And commented this in config.go better.

File: controllers/handler.go (hunk 3 above)
Comment (kalantar) [reply controllers/handler.go]: Done. Also did so for environment variable.

**Comment by kalantar** [comment]
@sriumcp All requested changes have been made.

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**